onnxruntime/core/optimizer/fast_gelu_fusion.cc

```diff
@@ -0,0 +1,168 @@
+    input_index = optimizer_utils::IndexOfNodeInput(add1_node, *mul2_node.MutableOutputDefs()[0]);
+    const float one = 1.0f;
+    if (!CheckNode(add1_node, "Add", {7}, mul1_node.GetExecutionProviderType(), true) ||
+        !optimizer_utils::IsInitializerWithExpectedValue(graph, *(add1_node.InputDefs()[(input_index + 1) % 2]), one, true)) {
+      continue;
+    }
+
+
+    Node& mul3_node = *graph.GetNode(add1_node.OutputNodesBegin()->Index());
+    if (!CheckNode(mul3_node, "Mul", {7}, mul1_node.GetExecutionProviderType(), true)) {
+      continue;
+    }
+
+
+    input_index = optimizer_utils::IndexOfNodeInput(mul3_node, *add1_node.MutableOutputDefs()[0]);
+    Node& mul4_node = const_cast<Node&>(*graph_utils::GetInputNode(mul3_node, (input_index + 1) % 2));
+    if (!CheckNode(mul4_node, "Mul", {7}, mul1_node.GetExecutionProviderType(), true)) {
+      continue;
+    }
+
+    input_index = -1;
+    const float mul4_val = 0.7978845834732056f;
+    for (auto i = 0; i < 2; i++) {
+      if (optimizer_utils::IsInitializerWithExpectedValue(graph, *(mul4_node.InputDefs()[i]), mul4_val, true)){
+        input_index = i;
+        break;
+      }
+    }
+
+    if (input_index == -1 || mul4_node.InputDefs()[(input_index + 1) % 2]->Name() != gelu_input_arg->Name())
+        continue;
+
+
+    Node& tanh_node = *graph.GetNode(mul3_node.OutputNodesBegin()->Index());
+    if (!CheckNode(tanh_node, "Tanh", {6}, mul1_node.GetExecutionProviderType(), true)) {
+      continue;
+    }
+
+
+    Node& add2_node = *graph.GetNode(tanh_node.OutputNodesBegin()->Index());
+    if (!CheckNode(add2_node, "Add", {7}, mul1_node.GetExecutionProviderType(), true)) {
+      continue;
+    }
+
+    input_index = optimizer_utils::IndexOfNodeInput(add2_node, *tanh_node.MutableOutputDefs()[0]);
+    if (!optimizer_utils::IsInitializerWithExpectedValue(graph, *(add2_node.InputDefs()[(input_index + 1) % 2]), one, true)) {
+      continue;
+    }
+
+
+    Node& mul5_node = *graph.GetNode(add2_node.OutputNodesBegin()->Index());
+    if (!CheckNode(mul5_node, "Mul", {7}, mul1_node.GetExecutionProviderType(), true)) {
+      continue;
+    }
+
+
+    input_index = optimizer_utils::IndexOfNodeInput(mul5_node, *add2_node.MutableOutputDefs()[0]);
+    Node& mul6_node = const_cast<Node&>(*graph_utils::GetInputNode(mul5_node, (input_index + 1) % 2));
+    if (!CheckNode(mul6_node, "Mul", {7}, mul1_node.GetExecutionProviderType(), false)) {
+      continue;
+    }
+
+    const float mul6_val = 0.5f;
+    input_index = -1;
+    for (auto i = 0; i < 2; i++) {
+      if (optimizer_utils::IsInitializerWithExpectedValue(graph, *(mul6_node.InputDefs()[i]), mul6_val, true)){
+        input_index = i;
+        break;
+      }
+    }
+
+    if (input_index == -1 || mul6_node.InputDefs()[(input_index + 1) % 2]->Name() != gelu_input_arg->Name())
+      continue;
+
+
+    const std::vector<NodeArg*> gelu_input_defs{mul1_node.MutableInputDefs()[0]};
```

Comment:
`FastGelu` supports an optional bias term (`Add` node at start).  Could this fusion transformer check for the starting `Add` node and optionally add it to inputs (2nd input)?

Reply:
See here:
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/contrib_ops/cuda/bert/fast_gelu.cc#L45

Reply:
oh yes! we should do that.